### ISSUE TITLE
[Task]: Update mypy type checker in tox.ini

### ISSUE THREAD

**Issue by AnandInguva** [apache/beam#25615]
### What needs to happen?

If we try to update mypy-protobuf in build-requirements.txt to a latest version and run myypy==0.782(this is what we use currently to run mypy) type checker on the stubs generated by mypy-protobuf, we will see some errors which are wrong. 

![image](https://user-images.githubusercontent.com/34158215/220982644-aa73e7ab-0841-4836-a17c-77e197c9b371.png)

Update mypy to the latest version since the current version we use (0.782) was released almost 2 years back.

### Issue Priority

Priority: 3 (nice-to-have improvement)

### Issue Components

- [X] Component: Python SDK
- [ ] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [ ] Component: IO connector
- [ ] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

**Comment by AnandInguva** [comment]
.take-issue

**Comment by AnandInguva** [comment]
Version 0.790 seems to solve this and raise 7 errors. The newer version 1.0.1 is throwing 128 errors on  the PR https://github.com/apache/beam/pull/24599

I will go ahead and update mypy to 0.790 to unblock https://github.com/apache/beam/pull/24599

**Comment by AnandInguva** [comment]
Version 0.790 seems to throw minimal errors with Beam codebase.  mypy 1.0.1 throwing 128 errors indicates that Beam code base has lot more type hinting errors which were not identified by old versions of mypy.
